synthtool/gcp/common.py

```diff
@@ -84,7 +82,7 @@ def node_library(self, **kwargs) -> Path:
             if "samples/README.md" not in self.excludes:
                 self.excludes.append("samples/README.md")
 
-        kwargs["metadata"] = node.read_metadata()
+        kwargs["metadata"] = node._template_metadata()
```

Comment:
In python, members named with a leading _ are intended to be private.  Why is this code calling a private method?

Reply:
The idea of naming node._template_metadata() was for it to be private for users of synthtool, not necessarily for within the package. If that's incorrect, then we can make it public.

Reply:
Hmmm.  The fact that common.py is importing specific code from node.py is a red flag, but I realize that precedes this pull request.

I'd rother see `_template_metadata` renamed `template_metadata`.

Reply:
It's not a huge deal, I was just worried about the public surface area of these modules in case people are using them directly in their synth files. For this case, it could be reasonable for someone to need access to these variables anyways.